### PR TITLE
Fix the events in StateChannel

### PR DESCRIPTION
**Pull request by nz-neooo**
After https://github.com/subquery/network-contracts/pull/319 tested on testnet, we will upgrade it on kepler for smaller fix. @mzxyz @ianhe8x

### CODE REVIEW

File: contracts/StateChannel.sol [hunk 1]
```diff
@@ -364,6 +364,13 @@ contract StateChannel is Initializable, OwnableUpgradeable {
         bool isClaimable1 = channels[channelId].status == ChannelStatus.Terminating &&
             channels[channelId].terminatedAt < block.timestamp;
 
+        // if channel is finalized, and status not change, just return
+        if (isClaimable1 && channels[channelId].indexer == address(0)) {
+            delete channels[channelId];
+            emit ChannelFinalize(channelId, 0, 0);
```
Comment (ianhe8x) [inline contracts/StateChannel.sol]: I think we shouldn't re-emit finalization event. Don't worry about subquery project, they can be fixed by adding some status check.

Comment (nz-neooo) [reply contracts/StateChannel.sol]: It’s better to keep it, prevent something that was not considered.

Comment (ianhe8x) [reply contracts/StateChannel.sol]: finalizing a channel in `claim` is very weird, and may override the real finalization time.
This event will be ignored by subquery project anyway, due to same reason.

Comment (mzxyz) [reply contracts/StateChannel.sol]: There also have a `_finalize` fun call at the end of claim fun which also emit Finalise event, seems it is alright to manually emit the finalise event in this condition, and it is not conflict with other cases.

**Comment by nz-neooo** [comment]
Does anyone know why the test failed?

File: contracts/StateChannel.sol (hunk 1 above)
Comment (nz-neooo) [reply contracts/StateChannel.sol]: we had remove this event, because issue will be fixed on subql & chs by hand.

**Review by mzxyz — APPROVED** [review]

**Pull request merged**